src/background.ts

```diff
@@ -16,22 +16,22 @@ chrome.storage.sync.get("color", (items) => {
  * used to compare when new tabs are created. It's used to compare the suffixes, that is
  * google.com will block both www.google.com and maps.google.com.
  */
-var blocked_hostnames = [];
+let blocked_hostnames: string[] = [];
 
 /**
  * an object where key is an tab id (integers) and value is count down period in seconds.
  * The tabs urls are in the above blocked_hostnames but are allowed
  * to navigate for a certain period of time.
  */
-var tabs_in_countdown = {};
+let tabs_in_countdown: Record<number, number> = {};
 
 chrome.storage.sync.get("blocked_hostnames", (host_suffixes) => {
   console.log("time locked urls", host_suffixes);
   // read into blocked_hostnames
   blocked_hostnames = host_suffixes.blocked_hostnames || [];
 });
 
-function is_hostname_blocked(url, tabId) {
+function is_hostname_blocked(url: string, tabId: number) {
```

Comment:
do we also annotate return type?

Reply:
For this function, it's safe to omit return type. TypeScript is smart enough to infer the return type.